### PR TITLE
Update QT to build on Intel

### PR DESCRIPTION
**Pull request by sethrj**
Update the correct qmake.conf files and use the correct `-platform` flag when using intel compilers.

### CODE REVIEW

File: var/spack/repos/builtin/packages/libmng/package.py [hunk 1]
```diff
@@ -25,11 +28,9 @@ def patch(self):
...
+    def cmake_args(self):
+        args = [('WITH_LCMS2', True),
+                ('WITH_LCMS1', False)]
 
-            configure: error: source directory already configured;
-            run "make distclean" there first
-        """
-        make('distclean')
+        return ['-D{0}:BOOL={1}'.format(key, 'ON' if val else 'OFF')
+                for (key, val) in args]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/libmng/package.py]: This seems more complex than it needs to be...

**Comment by adamjstewart** [comment]
Python change looks good. It would be easier to merge this if there were 3 separate PRs since the changes are unrelated.

File: var/spack/repos/builtin/packages/libmng/package.py (hunk 1 above)
Comment (sethrj) [reply var/spack/repos/builtin/packages/libmng/package.py]: Yes... when I first put it in, I had another option or two, and decided I'd format it for easy replacement with
```python
return [self.define(*kv) for kv in args]
```
pending  https://github.com/spack/spack/pull/14376 .

I can replace with an explicit set of arguments no problem.

**Comment by sethrj** [comment]
@adamjstewart Yeah I was going to ask about that. I'll break it into pieces.

**Comment by sethrj** [comment]
@adamjstewart This PR has been simplified to just the QT package changes and should be go for merge.

**Comment by sethrj** [comment]
Thanks @adamjstewart for the quick reviews and merges!

**Pull request merged**